lib/publisher.ex

```diff
@@ -9,6 +9,11 @@ defmodule GenRMQ.Publisher do
 
   require Logger
 
+  # list of fields permitted in message metadata at top level
+  @metadata_fields :P_basic
```

Comment:
not sure what `:P_basic` is but I like it :)

Reply:
There's a bunch of Erlang record definitions in `rabbit_common/include/rabbit_framing.hrl`, one is this:
```erlang
-record('P_basic', {content_type, content_encoding, headers, delivery_mode, priority, correlation_id, reply_to, expiration, message_id, timestamp, type, user_id, app_id, cluster_id}).
```
There isn't much documentation on it, but it is used everywhere in `amqp` to describe the metadata of a message. `Record.extract` translates that record definition to a keyword list.